grpc/server/src/plugins/mission/missionrpc_impl.h

```diff
@@ -9,11 +11,8 @@ using namespace dronecore;
 class MissionServiceImpl final : public rpc::mission::MissionService::Service
 {
 public:
-    MissionServiceImpl(DroneCore *dc_obj)
-    {
-        dc = dc_obj;
-        mission = std::make_shared<Mission>(&dc->device());
-    }
+    MissionServiceImpl(Mission &mission)
+        : mission(mission) {}
 
     Status SendMission(ServerContext *context, const rpc::mission::SendMissionRequest *request,
```

Comment:
Can we name it as `UploadMission()` (or just `Upload()` as its mission upload indeed, because we call as `mission.upload(...)`) ? Makes sense ?

Reply:
I don't have a strong opinion on it, but note that this implies changing the proto files, and very soon we won't be able to change the proto files anymore as it will break the API for all the modules.

This said, I had put `SendMission` instead of `UploadMission` because I remember having had weird discussion with users who were assuming that "upload" is over the internet, and hence you "upload the mission to some website" :D. But maybe that's not relevant here.

@julianoes do you have an opinion on the naming?

Reply:
Hm, when it is `SendMission` what's the other direction? `ReceiveMission`, `RequestMission`?

Reply:
Actually, I am completely wrong. It is already [UploadMission](https://github.com/dronecore/DroneCore-Proto/blob/master/mission/mission.proto#L9). It's just not in the scope of this PR :-). Sorry about that :-).